### PR TITLE
Keep publish secrets scoped to specific steps

### PR DESCRIPTION
**Pull request by armanbilge**
Currently, they are available in the entire workflow. This adds a new setting `githubWorkflowPublishEnv` that allows us to scope these secrets to only the publish job which should be the only place they are used.

Technically a breaking change if someone was relying on the old global scoping.

### CODE REVIEW

**Comment by armanbilge** [comment]
Actually, we can do even better here and keep them scoped to the specific steps maybe.

**Review by rossabaker — APPROVED** [review]
This doesn't reduce the malicious threat vector much, but it reduces the stupid threat vector, and that's a good thing.

**Comment by armanbilge** [comment]
Huh. I must have broke the workflows?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -132,17 +128,26 @@ jobs:
           rm targets.tar
 
       - name: Import signing key
-        if: env.PGP_SECRET != '' && env.PGP_PASSPHRASE == ''
+        if: secrets.PGP_SECRET != '' && secrets.PGP_PASSPHRASE == ''
```
Comment (armanbilge) [inline .github/workflows/ci.yml:131]: Secrets can't be used in conditionals. I'm not sure if we can use env here if we only set it for this step (not sure what happens first).

**Comment by armanbilge** [comment]
Not 100% convinced but let's see what happens on main.

**Pull request merged**